src/multiprocess/multiprocess_memory_limit.c

```
@@ -448,6 +449,75 @@ uint64_t nvml_get_device_memory_usage(const int dev) {
+            if (spins < SEQLOCK_SPIN_LIMIT) {
+                seqlock_cpu_relax();
+            } else if (spins < SEQLOCK_YIELD_LIMIT) {
+                sched_yield();
+            } else if (spins < SEQLOCK_SLEEP_LIMIT) {
+                usleep(100);
+            } else {
+                /* Nothing legitimate holds the slot this long, so the holder
+                 * died between begin and end. The slot is stuck odd and every
+                 * reader is already spinning on it, so step the counter back to
+                 * even and reclaim it rather than wedge the node. */
+                LOG_WARN("seqlock on slot pid %d stuck odd, reclaiming",
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)